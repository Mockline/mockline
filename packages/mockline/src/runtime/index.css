@import "tailwindcss";
@import "assets/css/animation.css";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";

@theme {
  --color-neutral: var(--ui-bg);
  --color-neutral-elevated: var(--ui-bg-elevated);
  --color-border: var(--ui-border);
}

@layer base {
  :root {
    color-scheme: light;

    --ui-text-dimmed: var(--ui-color-neutral-400);
    --ui-text-muted: var(--ui-color-neutral-500);
    --ui-text-toned: var(--ui-color-neutral-600);
    --ui-text: var(--ui-color-neutral-700);
    --ui-text-highlighted: var(--ui-color-neutral-900);

    --ui-bg: var(--color-white);
    --ui-bg-elevated: var(--ui-color-neutral-100);
    --ui-bg-accented: var(--ui-color-neutral-200);
    --ui-bg-inverted: var(--ui-color-neutral-900);

    --ui-border: var(--ui-color-neutral-200);
    --ui-border-accented: var(--ui-color-neutral-300);
    --ui-border-inverted: var(--ui-color-neutral-900);

    --ui-radius: var(--radius);

    /*--ui-slate: var(--color-slate-500);
    --ui-gray: var(--color-gray-500);
    --ui-zinc: var(--color-zinc-500);
    --ui-stone: var(--color-stone-500);
    --ui-neutral: var(--color-neutral-500);
    --ui-red: var(--color-red-500);
    --ui-orange: var(--color-orange-500);
    --ui-amber: var(--color-amber-500);
    --ui-yellow: var(--color-yellow-500);
    --ui-lime: var(--color-lime-500);
    --ui-green: var(--color-green-500);
    --ui-emerald: var(--color-emerald-500);
    --ui-teal: var(--color-teal-500);
    --ui-cyan: var(--color-cyan-500);
    --ui-sky: var(--color-sky-500);
    --ui-blue: var(--color-blue-500);
    --ui-indigo: var(--color-indigo-500);
    --ui-violet: var(--color-violet-500);
    --ui-purple: var(--color-purple-500);
    --ui-fuchsia: var(--color-fuchsia-500);
    --ui-pink: var(--color-pink-500);
    --ui-rose: var(--color-rose-500);*/
  }

  .dark {
    color-scheme: dark;

    --ui-text-dimmed: var(--ui-color-neutral-500);
    --ui-text-muted: var(--ui-color-neutral-400);
    --ui-text-toned: var(--ui-color-neutral-300);
    --ui-text: var(--ui-color-neutral-200);
    --ui-text-highlighted: var(--color-white);

    --ui-bg: var(--ui-color-neutral-900);
    --ui-bg-elevated: var(--ui-color-neutral-800);
    --ui-bg-accented: var(--ui-color-neutral-700);
    --ui-bg-inverted: var(--color-white);

    --ui-border: var(--ui-color-neutral-800);
    --ui-border-accented: var(--ui-color-neutral-700);
    --ui-border-inverted: var(--color-white);
  }
}

