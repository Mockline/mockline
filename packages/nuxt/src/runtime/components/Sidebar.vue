<script setup lang="ts">
import { ref } from 'vue'

type SidebarProps = {
  minSize?: number
  maxSize?: number
  collapsedSize?: number
  defaultSize?: number
}

const props = withDefaults(defineProps<SidebarProps>(), {
  minSize: 15,
  maxSize: 15,
  collapsedSize: 0,
  defaultSize: 20,
})

type SidebarSlots = {
  header: any
  default: any
  footer: any
}

const slots = defineSlots<SidebarSlots>()

const collapsed = ref(false)

const sidebarRef = ref<HTMLElement | null>(null)
</script>

<template>
  <nav ref="sidebarRef" class="flex flex-col h-full justify-between">
    <div class="flex flex-col gap-4">
      <div>
        <slot name="header" />
      </div>
      <div>
        <slot name="default" />
      </div>
    </div>
    <div>
      <slot name="footer" />
    </div>
  </nav>
</template>

<style scoped>

</style>
