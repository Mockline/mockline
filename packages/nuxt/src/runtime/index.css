@import "@mockline/themes";
@import "./mockline.css";
