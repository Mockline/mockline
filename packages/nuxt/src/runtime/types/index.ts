// Elements
export * from '../components/Button.vue'
export * from '../components/NavMenu.vue'

// Layout
export * from '../components/App.vue'
export * from '../components/Aside.vue'
export * from '../components/Card.vue'
export * from '../components/Header.vue'
export * from '../components/Main.vue'
export * from '../components/Page.vue'
export * from '../components/PageBody.vue'
export * from '../components/PageHeader.vue'

// Settings
export * from '../components/ThemeToggle.vue'

// Overlays
export * from '../components/Toasts.vue'

// Content
export * from '../components/content/ContentNavigationTree.vue'
export * from '../components/content/ContentSurround.vue'
export * from '../components/content/ContentToc.vue'
export * from '../components/content/ContentTocLinks.vue'

// Global
export * from '../components/prose/Callout.vue'
export * from '../components/prose/CodeGroup.vue'
export * from '../components/prose/ProseCode.vue'
export * from '../components/prose/CodeButton.vue'
