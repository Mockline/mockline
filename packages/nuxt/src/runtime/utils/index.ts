export function transformSlot(slot: any, index: number): any {
  if (typeof slot.type === 'symbol') {
    return slot.children?.map(transformSlot)
  }
  return {
    label: slot.props?.filename || slot.props?.label || `${index}`,
    icon: slot.props?.icon,
    component: slot
  }
}

export * from './useComponent'
