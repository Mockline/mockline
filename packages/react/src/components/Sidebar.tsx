"use client";

import React, {useState} from "react";
import {NavigationTreeLink} from "@mockline/themes";
import {MHeader} from "./Header.js";
import {MContentNavigationTree} from "./content/ContentNavigationTree.js";

type MSidebarProps = {
  links: NavigationTreeLink[];
  minSize?: number;
  maxSize?: number;
  collapsedSize?: number;
  defaultSize?: number;
  version?: string;
};

export const MSidebar: React.FC<MSidebarProps> = ({
                                                    links,
                                                    minSize = 200,
                                                    maxSize = 300,
                                                    collapsedSize = 60,
                                                    defaultSize = 250,
                                                    version = "1.0.0",
                                                  }) => {
  const [collapsed, setCollapsed] = useState(defaultSize <= collapsedSize);


  const toggleSidebar = () => {
    setCollapsed((prev) => !prev);
  };

  const sidebarWidth = collapsed ? `${collapsedSize}px` : `${maxSize}px`;

  return (
    <nav
      className={`flex flex-col h-screen bg-gray-800 text-white transition-all duration-300`}
      style={{width: sidebarWidth}}
    >
      {/* Header */}
      <MHeader className="flex items-center justify-between px-4 py-3 border-b border-gray-700">
        <span className="text-lg font-bold">Mockline</span>
        <button onClick={toggleSidebar}>
          <i className="text-lg lucide:menu"/>
        </button>
      </MHeader>

      {/* Navigation */}
      <MContentNavigationTree links={links} getClasses={(key: string, classes: Record<string, boolean>) => {
        if (classes?.active) {
          return "bg-gray-700";
        }
        if (classes?.disabled) {
          return "text-gray-400";
        }
        return "";
      }}/>

      {/* Footer */}
      <footer className="px-4 py-3 border-t border-gray-700">
        <span className="bg-gray-600 px-2 py-1 rounded text-xs">
          v{version}
        </span>
      </footer>


    </nav>
  );
};
