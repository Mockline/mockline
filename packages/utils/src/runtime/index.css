@import "tailwindcss";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";
