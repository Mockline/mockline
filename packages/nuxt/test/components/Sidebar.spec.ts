import { describe, it, expect } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'
import Sidebar from '../../src/runtime/components/Sidebar.vue'

describe('Sidebar', () => {
  it('can mount component', async () => {
    const html = await renderSuspended(Sidebar)
    expect(html.html()).toMatchSnapshot()
  })
})
