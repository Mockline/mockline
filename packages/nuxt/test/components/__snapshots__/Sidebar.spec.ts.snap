// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar > can mount component 1`] = `
"<div id="test-wrapper">
  <nav class="flex flex-col h-full justify-between px-3 py-2">
    <div class="flex flex-col gap-4">
      <div></div>
      <div></div>
    </div>
    <div></div>
  </nav>
</div>"
`;
